src/Microsoft.OData.Core/Value/ODataStreamReferenceValue.cs

```diff
@@ -8,21 +8,108 @@ namespace Microsoft.OData
 {
     #region Namespaces
     using System;
+    using System.IO;
     using Microsoft.OData.Evaluation;
+    using Edm;
 
     #endregion
 
+    // todo (mikep): move these to their own files
+
     /// <summary>
-    /// Represents a media resource.
+    /// A class to represent a binary stream value
     /// </summary>
-    public sealed class ODataStreamReferenceValue : ODataValue
+    public sealed class ODataBinaryStreamValue : ODataValue
     {
-        /// <summary>the metadata builder for this OData resource.</summary>
-        private ODataResourceMetadataBuilder metadataBuilder;
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="stream">Input stream</param>
+        public ODataBinaryStreamValue(Stream stream)
+        {
+            this.Stream = stream;
+        }
 
+        /// <summary>
+        /// The Stream wrapped by the ODataStreamValue
+        /// </summary>
+        public Stream Stream { get; private set; }
+    }
+
+    /// <summary>
+    /// A class to represent a stream value
+    /// </summary>
+    public class ODataStreamValue : ODataValue

```

Comment:
>ODataStreamValue [](start = 17, length = 16)

mark it as abstract class?

Reply:
Changed the hierarchy. This is not an ODataStreamInfo, which is sealed.

---
In reply to: [258748878](https://github.com/OData/odata.net/pull/1204#discussion_r258748878) [](ancestors = 258748878)